cmd/otel-allocator/config/flags.go

```diff
@@ -69,8 +69,13 @@ func getListenAddr(flagSet *pflag.FlagSet) (string, error) {
 	return flagSet.GetString(listenAddrFlagName)
 }
 
-func getPrometheusCREnabled(flagSet *pflag.FlagSet) (bool, error) {
-	return flagSet.GetBool(prometheusCREnabledFlagName)
+func getPrometheusCREnabled(flagSet *pflag.FlagSet) (value bool, changed bool, err error) {
+	if changed = flagSet.Changed(prometheusCREnabledFlagName); !changed {
```

Comment:
Because if it's not, we'll unconditionally replace the value from the config file with the default value from the flag, even if the flag is unset. We only want to give flags priority if they're actually set.